backend/fms_core/migrations/0065_v4_12_0.py

```diff
@@ -28,6 +28,73 @@ def set_measured_volume_properties_optional(apps, schema_editor):
+        admin_user = User.objects.get(username=ADMIN_USERNAME)
+        reversion.set_comment(f"Create Sample QC DNA and RNA step and transfer existing Sample QC data to their respective new step.")
+        reversion.set_user(admin_user)
+        for step_info in STEPS:
+            protocol = Protocol.objects.get(name=step_info["protocol_name"])
+            step = Step.objects.create(name=step_info["name"],
+                                       protocol_id=protocol.id,
+                                       type="PROTOCOL",
+                                       needs_placement=False,
+                                       needs_planning=False,
+                                       expected_sample_type=step_info["expected_sample_type"],
+                                       created_by_id=admin_user.id,
+                                       updated_by_id=admin_user.id)
+            reversion.add_to_revision(step)
+            for specification in step_info["specifications"]:
+                step_specification = StepSpecification.objects.create(name=specification["display_name"],
+                                                                      sheet_name=specification["sheet_name"],
+                                                                      column_name=specification["column_name"],
+                                                                      value=specification["value"],
+                                                                      step=step,
+                                                                      created_by_id=admin_user.id,
+                                                                      updated_by_id=admin_user.id)
+                reversion.add_to_revision(step_specification)
+        oldStep = Step.objects.get(name="Sample QC")
+        dnaStep = Step.objects.get(name="Sample QC (DNA)")
+        rnaStep = Step.objects.get(name="Sample QC (RNA)")
+        # Start for data transfer process
+        sns = SampleNextStep.objects.filter(step__id=oldStep.id)
+        for sampleNextStep in sns:
+            updatedSampleNextStep = SampleNextStep.objects.get(id=sampleNextStep.id)
+            if sampleNextStep.sample.derived_samples.first().sample_kind.name == "DNA":
+              updatedSampleNextStep.step = dnaStep
+            elif sampleNextStep.sample.derived_samples.first().sample_kind.name == "RNA":
+              updatedSampleNextStep.step = rnaStep
+            updatedSampleNextStep.save()
+            reversion.add_to_revision(updatedSampleNextStep)
+        # make sure there are no loose ends before deleting oldStep
+        so = StepOrder.objects.filter(step__id=oldStep.id)
+        for order in so:
+            if "Extraction (DNA)" in order.previous_step_order.first().step.name:
+                order.step = dnaStep
+            elif "Extraction (RNA)" in order.previous_step_order.first().step.name:
+                order.step = rnaStep
```

Comment:
Optional: similarly, I'm biased towards obtaining a `Step` model directly and comparing by id rather than name

Reply:
In fact, using `in` is more risky in general. It's preferable to use `==` or maybe even compare by step id if possible.

Reply:
well the distinction here is important, the string comparison is not merely there for fun, we need to know what type sequencing the previous step was doing and comparing by id does not give us this answer